core/src/main/java/com/linecorp/armeria/server/annotation/HttpFileResponseConverterFunction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.file.HttpFile;
+
+/**
+ * A response converter implementation which creates an {@link HttpResponse} when the {@code result} is
+ * an instance of {@link HttpFile}.
+ */
+public final class HttpFileResponseConverterFunction implements ResponseConverterFunction {
+
+    @Override
+    public HttpResponse convertResponse(ServiceRequestContext ctx,
+                                        ResponseHeaders headers,
+                                        @Nullable Object result,
+                                        HttpHeaders trailers) throws Exception {
+        if (result instanceof HttpFile) {
+            return ((HttpFile) result).asService().serve(ctx, ctx.request());
```

Comment:
I think this does not work when a user specify
`@AdditionalHeader` and `@AdditionalTrailer` because we are completely ignoring the `headers` and `trailers`.
How about using `FilteredHttpResponse` so that we can set the additional headers to the response?
And I'm sure that you know how to use `FilteredHttpResponse`, right? 😄

Reply:
How about adding `HttpFile.toBuilder()` for mutating it?

Reply:
I'm not sure if it's possible because there are different `HttpFile`s such as `DeferredHttpFile`, `CachingHttpFile`, etc.
But it might be worth trying. 😄

Reply:
Thanks for the comments.
I find it tricky implementing `toBuilder()` option as @minwoox has pointed out.

I will try both, though!